### PR TITLE
feat: add fn to check enterprise user

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Implement API to check for enterprise user

## Objective

This PR adds a function to determine if a user belongs to an enterprise

## References

https://github.com/screwdriver-cd/scm-github-graphql/pull/1

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1967,6 +1986,23 @@ class GithubScm extends Scm {
...
+     * @param {String} config.login  The login of the Github user
+     * @returns Boolean
+     */
+    async _isEnterpriseUser(config) {
+        if (!this.scmGithubGQL) {
+            return false;
+        }
+        const user = await this.scmGithubGQL.getEnterpriseUserAccount(config);
+
+        return !!user && user.type === ENTERPRISE_USER;
```
Comment (jithine) [inline index.js:2004]: does this verify if the enterprise user belongs, specific to yahoo enterprise account ?

Comment (jithine) [reply index.js:2004]: do we need to ? 🤔

Comment (pritamstyz4ever) [reply index.js:2004]: yes, it's generic and uses the configured enterprise slug for getting this data

File: test/index.test.js [hunk 2]
```diff
@@ -3354,4 +3359,61 @@ jobs:
             });
         });
     });
+
+    describe.only('_isEnterpriseUser', () => {
```
Comment (tkyi) [inline test/index.test.js:3363]:
```suggestion
    describe('_isEnterpriseUser', () => {
```

File: index.js [hunk 3]
```diff
@@ -236,7 +241,15 @@ class GithubScm extends Scm {
                     oauthClientId: joi.string().required(),
                     oauthClientSecret: joi.string().required(),
                     fusebox: joi.object().default({}),
-                    secret: joi.string().required()
+                    secret: joi.string().required(),
+                    gheCloud: joi
```
Comment (tkyi) [inline index.js:245]: do you need to add `gheCloudSlug` too?

**Review by tkyi — COMMENTED** [review]
do you need to update scm-base?

**Pull request merged**